vnext/Microsoft.ReactNative/ReactApplication.h

```diff
@@ -95,11 +97,61 @@ struct ReactApplication : NoDefaultCtorReactApplication_base<ReactApplication> {
   void OnSuspending(IInspectable const &, Windows::ApplicationModel::SuspendingEventArgs const &);
   void OnNavigationFailed(IInspectable const &, xaml::Navigation::NavigationFailedEventArgs const &);
 
+  using AppLaunchedDelegate = winrt::delegate<void(
+      winrt::Microsoft::ReactNative::ReactApplication sender,
+      winrt::Windows::ApplicationModel::Activation::LaunchActivatedEventArgs args)>;
+
+  void LaunchedInternal(AppLaunchedDelegate del) {
+    m_launched = del;
+  }
+  AppLaunchedDelegate LaunchedInternal() {
+    return m_launched;
+  }
+
+  using AppViewCreatedDelegate =
+      winrt::delegate<void(winrt::Microsoft::ReactNative::ReactApplication sender, winrt::hstring args)>;
+
+  void ViewCreatedInternal(AppViewCreatedDelegate del) {
+    m_viewCreated = del;
+  }
+
+  AppViewCreatedDelegate ViewCreatedInternal() {
+    return m_viewCreated;
+  }
+
+  using AppPageNavigatedDelegate =
+      winrt::delegate<void(winrt::Microsoft::ReactNative::ReactApplication sender, ReactRootView view)>;
```

Comment:
>ReactRootView [](http://example.com/codeflow?start=83&length=13)

We typically must not pass parameters with non-trivial destructors by value. It generates inefficient code.
In C++/WinRT we typically pass them by 'const &'.
The same is for all new functions in this file.